File: src/com/ashcraftmedia/daftman/sprite/Fire.java
package com.ashcraftmedia.daftman.sprite;

import java.awt.Graphics;
import java.awt.Image;

import com.ashcraftmedia.daftman.core.ImageStore;
import com.ashcraftmedia.daftman.scene.SceneDirector;




/**
 * HW10: DAFTMAN
 * I worked on this assignment alone, using course materials, previous work,
 * the online Java API.
 *  
 * Fire.java
 * This class represents a fire.
 * 
 * @author Ryan Ashcraft
 * @version 1.0 12/03/2010
 */

public class Fire extends Sprite {
	private static Image[] fireImages;
	
	protected FireDelegate delegate;
	final double STEP_SPEED_MULTIPLIER = 0.1;
	
	/**
	 * Constructor for Fire objects.
	 * 
	 * @param aDelegate The FireDelegate object
	 */
	public Fire(FireDelegate aDelegate) {
		delegate = aDelegate;
		
		fireImages = ImageStore.get().getAnimation("FIRE");
	}

	/**
	 * Draws a fire image.
	 * 
	 * @param g The Graphics object
	 */
	public void draw(Graphics g) {
		g.drawImage(fireImages[(int)(stepCount * STEP_SPEED_MULTIPLIER) % 2], loc.x, loc.y, size.width, size.height, null);
	}

	/**
	 * Eventually tells delegate to stop the fire after a number of steps.
	 */
	public void act() {	
		stepCount++;
		
		if (SceneDirector.get().secondsToCycles(1) == stepCount) {
			delegate.stopFire(this);
		}
	}
}

File: web/js/sprite/Fire.js
import Sprite from './Sprite.js';
import ImageStore from '../core/ImageStore.js';
import SceneDirector from '../scene/SceneDirector.js';

export default class Fire extends Sprite {
    constructor(aDelegate) {
        super();

        this.delegate = aDelegate;
        this.STEP_SPEED_MULTIPLIER = 0.1;
        this._fireImages = ImageStore.get().getAnimation('FIRE');
    }

    draw(ctx) {
        ctx.drawImage(
            this._fireImages[Math.floor(this.stepCount * this.STEP_SPEED_MULTIPLIER) % 2],
            this.loc.x,
            this.loc.y,
            this.size.width,
            this.size.height
        );
    }

    act() {
        this.stepCount++;

        if (SceneDirector.get().secondsToCycles(1) === this.stepCount) {
            this.delegate.stopFire(this);
        }
    }
}
